funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -151,50 +115,44 @@ def __init__(
         self.config = config
         log.info(f"Got config: {config}")
 
-        self.client_address = client_address
-        self.interchange_address = interchange_address
-        self.client_ports = client_ports
-        self.suppress_failure = suppress_failure
-
         self.endpoint_dir = endpoint_dir
         self.endpoint_name = endpoint_name
 
         if funcx_client_options is None:
             funcx_client_options = {}
+        # off_process checker is only needed on client side
+        funcx_client_options["use_offprocess_checker"] = False
         self.funcx_client = FuncXClient(**funcx_client_options)
 
         self.initial_registration_complete = False
+        self.connection_params = None
         if reg_info:
+            self.connection_params = reg_info
             self.initial_registration_complete = True
-            self.apply_reg_info(reg_info)
 
         self.heartbeat_period = self.config.heartbeat_period
         self.heartbeat_threshold = self.config.heartbeat_threshold
         # initalize the last heartbeat time to start the loop
         self.last_heartbeat = time.time()
-        self.keys_dir = keys_dir
         self.serializer = FuncXSerializer()
 
-        self.pending_task_queue = Queue()
+        self.pending_task_queue = multiprocessing.Queue()
         self.containers = {}
         self.total_pending_task_count = 0
 
         self._quiesce_event = threading.Event()
+        self._quiesce_complete = threading.Event()
         self._kill_event = threading.Event()
```

Comment:
I think these should be `multiprocessing.Event`?

stdlib docs make it unclear if they share an implementation, but I'd expect that we there's some distinction.